staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config_test.go

```diff
@@ -646,6 +648,187 @@ func TestKMSPluginHealthz(t *testing.T) {
 	}
 }
 
+// tests for masking rules
+func TestWildcardMasking(t *testing.T) {
+	testCases := []struct {
+		desc          string
+		config        *apiserverconfig.EncryptionConfiguration
+		expectedError bool
+		errorString   string
+	}{
+		{
+			desc: "resources masked by *.group",
+			config: &apiserverconfig.EncryptionConfiguration{
+				Resources: []apiserverconfig.ResourceConfiguration{
+					{
+						Resources: []string{
+							"configmaps",
+							"*.",
+							"secrets",
+						},
+						Providers: []apiserverconfig.ProviderConfiguration{
+							{
+								KMS: &apiserverconfig.KMSConfiguration{
+									Name:       "kms",
+									APIVersion: "v1",
+									Timeout:    &metav1.Duration{Duration: 3 * time.Second},
+									Endpoint:   "unix:///tmp/testprovider.sock",
+									CacheSize:  pointer.Int32(10),
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedError: true,
+			errorString:   fmt.Sprintf("resource secrets is masked by earlier rule %s", schema.GroupResource{Group: "", Resource: "*"}),
```

Comment:
Do not use `fmt.Sprintf` here - write out the actual error so it easy for us to see the actual text a user would see.